projects/hipsparse/CMakeLists.txt

```
@@ -1,289 +1,278 @@
 endif()
-set(CPACK_RESOURCE_FILE_LICENSE "${CMAKE_CURRENT_SOURCE_DIR}/LICENSE.md")
+
+# --- Client targets ---
+
+if(HIPSPARSE_BUILD_CLIENTS_ONLY OR HIPSPARSE_ENABLE_CLIENT OR HIPSPARSE_BUILD_TESTING OR BUILD_TESTING)
+    if(NOT CLIENTS_OS)
+        rocm_set_os_id(CLIENTS_OS)
+        string(TOLOWER "${CLIENTS_OS}" CLIENTS_OS)
+        rocm_read_os_release(CLIENTS_OS_VERSION VERSION_ID)
+    endif()
+    message(STATUS "OS: ${CLIENTS_OS} ${CLIENTS_OS_VERSION}")
+    
+    set(GFORTRAN_RPM "libgfortran4")
+    set(GFORTRAN_DEB "libgfortran4")
+    
+    if(CLIENTS_OS STREQUAL "centos" OR CLIENTS_OS STREQUAL "rhel" OR CLIENTS_OS STREQUAL "almalinux")
+        if(CLIENTS_OS_VERSION VERSION_GREATER_EQUAL "8")
+            set(GFORTRAN_RPM "libgfortran")
+        endif()
+    elseif(CLIENTS_OS STREQUAL "ubuntu" AND CLIENTS_OS_VERSION VERSION_GREATER_EQUAL "20.04")
+        set(GFORTRAN_DEB "libgfortran5")
+    elseif(CLIENTS_OS STREQUAL "mariner" OR CLIENTS_OS STREQUAL "azurelinux")
+        set(GFORTRAN_RPM "gfortran")
+    endif()
+    
+    if(HIPSPARSE_ENABLE_OPENMP)
+        set(OPENMP_RPM "libgomp")
+        set(OPENMP_DEB "libomp-dev")
+        if(CLIENTS_OS STREQUAL "sles")
+            set(OPENMP_RPM "libgomp1")
+        endif()
+    else()
+        set(OPENMP_RPM "")
+        set(OPENMP_DEB "")
+    endif()
+    
+    add_subdirectory(clients)
+endif()
+
+# --- Packaging and installation ---
+
+if(HIPSPARSE_BUILD_CLIENTS_ONLY OR HIPSPARSE_ENABLE_CLIENT OR HIPSPARSE_BUILD_TESTING OR BUILD_TESTING)
+    rocm_package_setup_component(clients)
+    rocm_package_setup_client_component(clients-common)
+    
+    if(HIPSPARSE_BUILD_TESTING OR BUILD_TESTING)
+        rocm_package_setup_client_component(
+            tests
+            DEPENDS
+            COMPONENT clients-common
+            DEB "${GFORTRAN_DEB}"
+            RPM "${GFORTRAN_RPM}"
+        )
+        rocm_package_setup_client_component(
+            tests
+            DEPENDS
+            COMPONENT clients-common
+            DEB "${OPENMP_DEB}"
+            RPM "${OPENMP_RPM}"
+        )
+    endif()
+    
+    if(HIPSPARSE_ENABLE_BENCHMARKS)
+        rocm_package_setup_client_component(
+            benchmarks
+            DEPENDS
+            COMPONENT clients-common
+            DEB "${OPENMP_DEB}"
+            RPM "${OPENMP_RPM}"
+        )
+    endif()
+    
+    if(NOT WIN32)
+        if(HIPSPARSE_BUILD_TESTING OR BUILD_TESTING)
+            rocm_package_add_rpm_dependencies(COMPONENT tests DEPENDS "${GFORTRAN_RPM}")
+            rocm_package_add_deb_dependencies(COMPONENT tests DEPENDS "${GFORTRAN_DEB}")
+        endif()
+        if(HIPSPARSE_ENABLE_BENCHMARKS)
+            rocm_package_add_rpm_dependencies(COMPONENT benchmarks DEPENDS "${GFORTRAN_RPM}")
+            rocm_package_add_deb_dependencies(COMPONENT benchmarks DEPENDS "${GFORTRAN_DEB}")
+        endif()
+        if(HIPSPARSE_ENABLE_OPENMP AND (HIPSPARSE_BUILD_TESTING OR BUILD_TESTING))
```

```suggestion
        if(HIPSPARSE_ENABLE_OPENMP)
```
This would allow for inconsistent configurations - i.e. I could turn off openmp and turn testing on and still end up here which seems wrong. If testing requires openmp we should error or warn depending on what previous behavior was.